pkg/platform/aws/aws.go

```diff
@@ -47,126 +43,99 @@ type workerPool struct {
-	OSVersion                string            `hcl:"os_version,optional"`
-	DNSZone                  string            `hcl:"dns_zone"`
-	DNSZoneID                string            `hcl:"dns_zone_id"`
-	ExposeNodePorts          bool              `hcl:"expose_nodeports,optional"`
-	SSHPubKeys               []string          `hcl:"ssh_pubkeys"`
-	CredsPath                string            `hcl:"creds_path,optional"`
-	ControllerCount          int               `hcl:"controller_count,optional"`
-	ControllerType           string            `hcl:"controller_type,optional"`
-	ControllerCLCSnippets    []string          `hcl:"controller_clc_snippets,optional"`
-	Region                   string            `hcl:"region,optional"`
-	EnableAggregation        bool              `hcl:"enable_aggregation,optional"`
-	DiskSize                 int               `hcl:"disk_size,optional"`
-	DiskType                 string            `hcl:"disk_type,optional"`
-	DiskIOPS                 int               `hcl:"disk_iops,optional"`
-	NetworkMTU               int               `hcl:"network_mtu,optional"`
-	HostCIDR                 string            `hcl:"host_cidr,optional"`
-	PodCIDR                  string            `hcl:"pod_cidr,optional"`
-	ServiceCIDR              string            `hcl:"service_cidr,optional"`
-	ClusterDomainSuffix      string            `hcl:"cluster_domain_suffix,optional"`
-	EnableReporting          bool              `hcl:"enable_reporting,optional"`
-	CertsValidityPeriodHours int               `hcl:"certs_validity_period_hours,optional"`
-	WorkerPools              []workerPool      `hcl:"worker_pool,block"`
+type flatcar struct {
+	OSName string `hcl:"os_name,optional"`
 }
 
-// init registers aws as a platform
-func init() {
-	platform.Register("aws", NewConfig())
+type network struct {
+	HostCIDR string `hcl:"host_cidr"`
 }
 
-func (c *config) LoadConfig(configBody *hcl.Body, evalContext *hcl.EvalContext) hcl.Diagnostics {
-	if configBody == nil {
-		return hcl.Diagnostics{}
-	}
-
-	if diags := gohcl.DecodeBody(*configBody, evalContext, c); len(diags) != 0 {
-		return diags
-	}
+type controller struct {
+	Type        string   `hcl:"type,optional"`
+	CLCSnippets []string `hcl:"clc_snippets,optional"`
+}
 
-	return c.checkValidConfig()
+type disk struct {
+	Size int    `hcl:"size,optional"`
+	Type string `hcl:"type,optional"`
+	IOPS int    `hcl:"iops,optional"`
 }
 
-func NewConfig() *config {
-	return &config{
-		Region:            "eu-central-1",
-		EnableAggregation: true,
-	}
+type config struct {
+	Metadata        *configpkg.Metadata
+	DNSZone         string       `hcl:"dns_zone"`
+	DNSZoneID       string       `hcl:"dns_zone_id"`
+	ExposeNodePorts bool         `hcl:"expose_nodeports,optional"`
+	CredsPath       string       `hcl:"creds_path,optional"`
+	Region          string       `hcl:"region,optional"`
+	WorkerPools     []workerPool `hcl:"worker_pool,block"`
+	Disk            *disk        `hcl:"disk,block"`
```

Comment:
It seems we should move controllers config into separate block, which would also use `workerPool` struct underneath.